sklearn/model_selection/_split.py

```diff
@@ -274,13 +274,38 @@ class _BaseKFold(BaseCrossValidator, metaclass=ABCMeta):
     """Base class for KFold, GroupKFold, and StratifiedKFold"""
 
     @abstractmethod
-    def __init__(self, n_splits, *, shuffle, random_state):
+    def __init__(
+        self,
+        n_splits,
+        *,
+        shuffle,
+        random_state,
+        x_shape=None,
+        max_train_size=None,
+        test_size=None,
+    ):
+        if isinstance(n_splits, int):
```

Comment:
Thinking about it, I think that we can keep the base class as is and make the base class accept a `str` without raising a warning.

Then, we can just specialize the `get_n_splits` for the `TimeSeriesSplit` that should not return directly `self.n_splits` but instead make the computation of `n_splits` if a string is provided.

Therefore, we only specialize the `TimeSeriesSplit` class.

Reply:
OK, I've replace the previous method with this `get_n_splits` method now.